### ISSUE TITLE
Possible regression for `Layout/BlockAlignment` cop

### ISSUE THREAD

**Issue by viralpraxis** [rubocop/rubocop#15312]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Layout/BlockAlignment

### Steps to reproduce the problem

```ruby
retry_on Exception,
  Exception2,
  wait: :polynomially_longer do |_, error|
    $stdout.puts(error)
  end
```

`EnforcedStyleAlignWith: start_of_block`

### Expected behavior

I expect this code to be offense-free, just as it was before https://github.com/rubocop/rubocop/commit/c6a3518126106fd8eaa6ff82b3663b5ebe6a4510.

### Actual behavior

```shell
$ cat test.rb | bundle exec rubocop --only Layout -A --stdin bug.rb
Inspecting 1 file
C

Offenses:

bug.rb:4:1: C: [Corrected] Layout/IndentationWidth: Use 2 (not 4) spaces for indentation. (https://rubystyle.guide#spaces-indentation)
    $stdout.puts(error)
^^^^
bug.rb:5:3: C: [Corrected] Layout/BlockAlignment: end at 5, 2 is not aligned with retry_on Exception, at 1, 0.
  end
  ^^^

1 file inspected, 2 offenses detected, 2 offenses corrected
====================
retry_on Exception,
  Exception2,
  wait: :polynomially_longer do |_, error|
  $stdout.puts(error)
end
```

### RuboCop configuration

```yaml
Layout/BlockAlignment:
  Description: Align block ends correctly.
  Enabled: true
  VersionAdded: '0.53'
  EnforcedStyleAlignWith: start_of_block
  SupportedStylesAlignWith:
  - either
  - start_of_block
  - start_of_line
```

### RuboCop version

```shell
1.88.0
```
